### ISSUE TITLE
MSSQL Doesn't populate the table references consistently

### ISSUE THREAD

**Issue by dlredden** [sequelize/sequelize-auto#349]
Given this table:
```sql
CREATE TABLE es.RackTurn
(
	RackTurnId BIGINT IDENTITY(1,1) NOT NULL
	, RackId INT NOT NULL
	, TurnId BIGINT NOT NULL
	, CONSTRAINT PK_RackTurn PRIMARY KEY CLUSTERED (RackTurnId)
) ON [PRIMARY];

ALTER TABLE es.RackTurn ADD CONSTRAINT FK_RackTurn_Rack
	FOREIGN KEY (RackId) REFERENCES es.Rack(RackId);

ALTER TABLE es.RackTurn ADD CONSTRAINT FK_RackTurn_Turn
	FOREIGN KEY (TurnId) REFERENCES es.Turn(TurnId);
```
This is the Sequelize model that's generated:
```js
module.exports = function(sequelize, DataTypes) {
  return sequelize.define(
    'rackTurn',
    {
      rackTurnId: {
        type: DataTypes.BIGINT,
        allowNull: false,
        primaryKey: true,
        field: 'RackTurnId',
      },
      rackId: {
        type: DataTypes.INTEGER,
        allowNull: false,
        primaryKey: true,
        field: 'RackId',
      },
      turnId: {
        type: DataTypes.BIGINT,
        allowNull: false,
        primaryKey: true,
        field: 'TurnId',
      },
    },
    {
      tableName: 'RackTurn',
      timestamps: false,
    }
  );
};
```
Notice the missing references, auto increment, schema, and the fact that FKs are mislabeled as PKs. I believe the definition should be as follows:
```js
module.exports = function(sequelize, DataTypes) {
  return sequelize.define(
    'rackTurn',
    {
      rackTurnId: {
        type: DataTypes.BIGINT,
        allowNull: false,
        primaryKey: true,
        autoIncrement: true,
        field: 'RackTurnId',
      },
      rackId: {
        type: DataTypes.INTEGER,
        allowNull: false,
        references: {
          model: 'rack',
          schema: 'es',
          key: 'RackId',
        },
        field: 'RackId',
      },
      turnId: {
        type: DataTypes.BIGINT,
        allowNull: false,
        references: {
          model: 'turn',
          schema: 'es',
          key: 'TurnId',
        },
        field: 'TurnId',
      },
    },
    {
      schema: 'es',
      tableName: 'RackTurn',
      timestamps: false,
    }
  );
};
```
Mildly related to #348 in that the schema issue is mentioned here but not part of the issue being reported by this ticket.

**Comment by dlredden** [comment]
I think the root of this issue lies in https://github.com/sequelize/sequelize-auto/blob/master/lib/dialects/index.js line 201. I think it should be a LEFT join but I've not proven that sequelize-auto can handle the null values that'll be returned by this change.

**Comment by steveschmitt** [comment]
Fixed in `sequelize-auto@0.5.1`
